Refactor and docs (#3)
rwjblue/ember-test-waiters: README.md

# ember-test-waiters

> This addon provides APIs to allow Ember testing to play nicely with other asynchronous

Reviewer: ```suggestion
This addon provides APIs to allow [@ember/test-helpers](https://github.com/emberjs/ember-test-helpers/) to play nicely with other asynchronous
```

events, such as an application that is waiting for a CSS3
transition or an IndexDB transaction. Waiters runs periodically
after each async helper (i.e. `click`, `andThen`, `visit`, etc) has executed,

> until a predetermined condition is met. After the waiters finish, the next async helper

Reviewer: Hmm, this isn't _quite_ correct. Can we rephrase to make it clearer that the helpers themselves return a promise, and the "settledness" of these test waiters is part of the condition that determines when that promise resolves?

Author: Sure, but to be clear, this is taken verbatim from here:

https://github.com/emberjs/ember.js/blob/master/packages/ember-testing/lib/test/waiters.js#L8

Author: Updated. Let me know what you think.

is executed and the process repeats.

This allows the test suite to pause at deterministic intervals, and helps thread together
the async nature of tests.

## Compatibility

- Ember.js v2.18 or above
- Ember CLI v2.13 or above

## Installation

```
ember install ember-test-waiters
```

## Usage

> Ember test waiters uses a minimal API to provide waiting functionality. This minimal API can be composed to accomodate various complex scenarios.

Reviewer: ```suggestion
`ember-test-waiters` uses a minimal API to provide waiting functionality. This minimal API can be composed to accommodate various complex scenarios.
```

### TestWaiter class

The `TestWaiter` class is, in most cases, all you will need to wait for async operations to complete before continuing tests.

```
import Component from '@ember/component';
import { TestWaiter } from 'ember-test-waiters';

> if (DEBUG) {

Reviewer: ```suggestion
let waiter = DEBUG && new TestWaiter('friend-waiter');
```

>   let waiter = new TestWaiter('friend-waiter');

Reviewer: ```suggestion
```

> }

Reviewer: ```suggestion
```
